### PR TITLE
Add string-hash definition

### PR DESCRIPTION
**Pull request by ethanrubio**
Please fill in this template.

- [x] Prefer to make your PR against the `types-2.0` branch.
- [x] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [x] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header.

### CODE REVIEW

File: string-hash/string-hash-tests.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/// <reference path="index.d.ts" />
+
+import * as hash from "string-hash";
+
+const newHash: number = hash("Mary had a little lamb."); // 1766333550
+
+if (newHash === 1766333550) {
```
Comment (andy-ms) [inline string-hash/string-hash-tests.ts]: Don't need the rest of the tests -- it takes a string, it returns a number, you're done.

File: string-hash/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "strictNullChecks": false,
```
Comment (andy-ms) [inline string-hash/tsconfig.json]: Make this `true`.

File: string-hash/string-hash-tests.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/// <reference path="index.d.ts" />
```
Comment (andy-ms) [inline string-hash/string-hash-tests.ts]: Don't need the reference -- the tsconfig.json will get it for you.

File: string-hash/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for string-hash 1.1.1
+// Project: https://github.com/darkskyapp/string-hash
+// Definitions by: Ethan Rubio <https://github.com/ethanrubio/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace StringHash {
```
Comment (andy-ms) [inline string-hash/index.d.ts]: This whole definition could just be:

```ts
function stringHash(str: string): number;
export = stringHash;
export as namespace stringHash;
```

(without any `declare module` wrapper)

Comment (andy-ms) [inline string-hash/index.d.ts]: Actually, looking at the [code](https://github.com/darkskyapp/string-hash/blob/master/index.js) this is not made available as a global variable. So you can leave off the `export as namespace` too.

Comment (ethanrubio) [inline string-hash/index.d.ts]: I'm getting a tsc error when I did what you recommended:

`'declare' modifier required for top level element`

Is the following acceptable instead?
```js
declare var stringHash: (str: string) => number;
export = stringHash;
```

Comment (andy-ms) [inline string-hash/index.d.ts]: That error just means you should put `declare` in front of `function`. (Same goes if you see this error in front of `namespace` or `class`).

**Review by ghost — APPROVED** [review]

**Pull request merged**